### PR TITLE
feat: [PLG-2123] Add version support for Feature Flags in Release Notes

### PR DESCRIPTION
**Pull request by neelam-harness**
## Describe your changes
Tested locally, works as expected.
Screen Shot 2023-06-14 at 10 57 08 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48685)

### CODE REVIEW

**Comment by neelam-harness** [comment]
trigger uts

**Review by kapil-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48685)* status: 2 file(s) to review. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/moduleversioninfo/runnable/UpdateVersionInfoTask.java` line 107 at r1](https://reviewable.io/reviews/harness/harness-core/48685#-NXyQSKL2r4lUMSy4vw5:-NXyQSKL2r4lUMSy4vw6:b-c8uxbs) ([raw file](https://github.com/harness/harness-core/blob/acbca62b7472f22bf0a3f3e198177564d6747be2/120-ng-manager/src/main/java/io/harness/ng/moduleversioninfo/runnable/UpdateVersionInfoTask.java#L107)):*
> ```Java
>     } else if (ModuleType.CF.name().equals(moduleName)) {
>       StringBuilder ffApiUrl = new StringBuilder();
>       ffApiUrl.append(nextGenConfiguration.getFfServerClientConfig().getBaseUrl()).append(FEATURE_FLAGS_ENDPOINT);
> ```

nit: I think we can concat a normal string as well, why need a stringBuilder for it?

**Review by ribeirophillipe — APPROVED** [review]

File: 120-ng-manager/src/main/resources/mvi/baseinfo.json [hunk 1]
```diff
@@ -40,9 +40,9 @@
      ]
    },
    {
-     "moduleName": "FF",
+     "moduleName": "CF",
```
Comment (davejohnston) [inline 120-ng-manager/src/main/resources/mvi/baseinfo.json:43]: Will this be displayed externally?  (Will the customer see it?).  Internally we have CF in alot of places, but externally it should be FF I believe.

**Comment by neelam-harness** [comment]
trigger review

**Comment by neelam-harness** [comment]
trigger messagecheck

File: 120-ng-manager/src/main/resources/mvi/baseinfo.json (hunk 1 above)
Comment (neelam-harness) [reply 120-ng-manager/src/main/resources/mvi/baseinfo.json:43]: Externally it's the full name that's displayed, so it won't show up as FF.
image

**Pull request merged**